homeassistant/components/samsungtv/media_player.py

```diff
@@ -77,33 +89,38 @@ async def async_setup_entry(hass, entry, async_add_entities):
 class SamsungTVDevice(MediaPlayerEntity):
     """Representation of a Samsung TV."""
 
-    def __init__(self, bridge, config_entry, on_script):
+    def __init__(
+        self,
+        bridge: SamsungTVLegacyBridge | SamsungTVWSBridge,
+        config_entry: ConfigEntry,
+        on_script: Script | None,
+    ) -> None:
         """Initialize the Samsung device."""
         self._config_entry = config_entry
-        self._host = config_entry.data[CONF_HOST]
-        self._mac = config_entry.data.get(CONF_MAC)
-        self._manufacturer = config_entry.data.get(CONF_MANUFACTURER)
-        self._model = config_entry.data.get(CONF_MODEL)
-        self._name = config_entry.data.get(CONF_NAME)
+        self._host: str | None = config_entry.data[CONF_HOST]
+        self._mac: str | None = config_entry.data.get(CONF_MAC)
+        self._manufacturer: str | None = config_entry.data.get(CONF_MANUFACTURER)
+        self._model: str | None = config_entry.data.get(CONF_MODEL)
+        self._name: str | None = config_entry.data.get(CONF_NAME)
         self._on_script = on_script
         self._uuid = config_entry.unique_id
         # Assume that the TV is not muted
-        self._muted = False
+        self._muted: bool = False
         # Assume that the TV is in Play mode
-        self._playing = True
-        self._state = None
+        self._playing: bool = True
+        self._state: str | None = None
         # Mark the end of a shutdown command (need to wait 15 seconds before
         # sending the next command to avoid turning the TV back ON).
-        self._end_of_power_off = None
+        self._end_of_power_off: datetime | None = None
         self._bridge = bridge
         self._auth_failed = False
         self._bridge.register_reauth_callback(self.access_denied)
 
-    def access_denied(self):
+    def access_denied(self) -> None:
         """Access denied callback."""
         LOGGER.debug("Access denied in getting remote object")
         self._auth_failed = True
-        self.hass.add_job(
+        self.hass.async_create_task(
```

Comment:
Why did we change this? As far as I can tell, this is called from sync context. We're not allowed to use the async API in sync context.

Reply:
Good catch. That's not good.

I expect it was probably because `add_job` will result in `homeassistant/components/samsungtv/media_player.py:124: error: Argument 1 to "add_job" of "HomeAssistant" has incompatible type "Coroutine[Any, Any, FlowResult]"; expected "Callable[..., Any]"  [arg-type]`

We should use `self.hass.create_task` here instead.  cc @chemelli74